### DISCUSSION TITLE
v2.0.0RC1

### DISCUSSION THREAD

**Discussion by 9seconds** [9seconds/mtg#182] [General]
A complete rewrite of the project. Please check the reasoning here: https://github.com/9seconds/mtg#version-2

<hr /><em>This discussion was created from the release <a href='https://github.com/9seconds/mtg/releases/tag/v2.0.0-rc1'>v2.0.0RC1</a>.</em>

**Comment by arisudesu** [comment]
Good job.

I'm going to try new version. I have a question in regard to config. Is the option to configure with environment variables removed completely, or I misread the readme? Is it possible to allow both file and env config, in similar to [what implemented in grafana](https://grafana.com/docs/grafana/latest/administration/configuration/#configure-with-environment-variables), restic, rclone and others? It is not very convenient to keep the configuration file on disk at all times when using Docker, and this is what I liked about the first version of the proxy - it could be launched directly from the image.

**Reply by 9seconds** [reply]
Thanks. Yes, this is a concern I’ve got a couple of time. From developer perspective, this is easier for me to have a structured config when cli implies flat one. I have worked with both approaches for some time and I feel myself better with configs. 

At the same time, I see your point and it is absolutely valid. I also value tools which run only with CLI and require nothing on FS to starup. 

My idea is that I’ll make a separate CLI command which accepts all config values with options or environment but I plan to do it in the next release. Say, 2.0.1. I guess, we can take best from both worlds: structured config, config file and stateless configuration with options and variables. 

What do you think?

**Reply by 9seconds** [reply]
https://github.com/9seconds/mtg/pull/198
